### PR TITLE
Fix gasslift reduction when under LRAT, GRAT or WRAT

### PR DESCRIPTION
**Pull request by totto82**

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

File: opm/simulators/wells/GasLiftSingleWellGeneric.cpp [hunk 1]
```diff
@@ -805,30 +805,68 @@ reduceALQtoOilTarget_(double alq,
...
+        gas_rate = -potentials[this->gas_pos_];
+        if (oil_is_limited && this->controls_.hasControl(Well::ProducerCMode::ORAT) && oil_rate < this->controls_.oil_rate) {
+            limiting_control = Well::ProducerCMode::ORAT;
+            break;
+        }
+        if (oil_is_limited && this->controls_.hasControl(Well::ProducerCMode::LRAT) && (oil_rate + water_rate) < this->controls_.liquid_rate) {
+            limiting_control = Well::ProducerCMode::LRAT;
+            break;
+        }
+        if (gas_is_limited && gas_rate < this->controls_.gas_rate) {
```
Comment (hakonhagland) [inline opm/simulators/wells/GasLiftSingleWellGeneric.cpp:818]: Shouldn't there be a check if the well has a GRAT control here? Similar to what is done above on line 810 and 814 for ORAT and LRAT..

Comment (totto82) [reply opm/simulators/wells/GasLiftSingleWellGeneric.cpp:818]: Since gas is limited we now that it has a GRAT control. I could add a assert.

**Comment by alfbr** [comment]
I have now tested, and confirm that this fixes the bug. Thank you!

**Pull request merged**